components/script/dom/html/htmlinputelement.rs

```diff
@@ -1420,17 +1518,85 @@ impl HTMLInputElement {
         );
     }
 
+    fn update_range_shadow_tree(&self, can_gc: CanGc) {
+        // Should only do this for `type=range` input.
+        debug_assert_eq!(self.input_type(), InputType::Range);
+
+        if self.owner_document().has_script_or_layout_blocker() {
+            let input_element = DomRoot::from_ref(self);
+            self.owner_document().add_delayed_task(task!(
+                ThumbPositionUpdate: |input_element: DomRoot<HTMLInputElement>| {
+                    input_element.update_range_position(CanGc::note());
+                }
+            ));
+            return;
+        }
+        self.update_range_position(can_gc);
+    }
+
+    fn update_range_position(&self, can_gc: CanGc) {
+        let mut value = self.Value();
+        if value.is_valid_floating_point_number_string() {
+            value.make_ascii_lowercase();
+        } else {
+            value = DOMString::from("0");
+        }
+        let min = self.minimum().unwrap_or(0.0);
+        let max = self.maximum().unwrap_or(100.0);
+        let value_num = self.convert_string_to_number(&value.str()).unwrap_or(0.0);
+        let clamped_value = if value_num < min {
+            min
+        } else if value_num > max {
+            max
+        } else {
+            value_num
+        };
```

Comment:
Couldn't this just be:


```suggestion
        let clamped_value = value_num.clamp(min, max);
```

?